### PR TITLE
Create initial specs integrating devices from ndx-ophys-devices

### PR DESCRIPTION
**Pull request by alessandratrapani**
see https://github.com/catalystneuro/ndx-ophys-devices/pull/1 for ophys devices
```mermaid
%%{init: {'theme': 'base', 'themeVariables': {'primaryColor': '#ffffff', "primaryBorderColor': '#144E73', 'lineColor': '#D96F32'}}}%%


classDiagram
    direction BT

    class MicroscopySeries {
        <<TimeSeries>>

        --------------------------------------
        datasets
        --------------------------------------
        microscopy_table_region : DynamicTableRegion
    }

    class PlanarMicroscopySeries {
        <<MicroscopySeries>>

        --------------------------------------
        datasets
        --------------------------------------
        data : numeric, frame x height x width
        --> unit : text

        --------------------------------------
        links
        --------------------------------------
        imaging_space : PlanarImagingSpace
    }

    class VariableDepthMicroscopySeries {
        <<MicroscopySeries>>

        --------------------------------------
        datasets
        --------------------------------------
        data : numeric, frame x height x width
        --> unit : text
        depth_per_frame : numeric, length of frames
        -- > unit : text, default="micrometers"

        --------------------------------------
        links
        --------------------------------------
        imaging_space : PlanarImagingSpace
    }

    class VolumetricMicroscopySeries {
        <<MicroscopySeries>>

        --------------------------------------
        datasets
        --------------------------------------
        data : numeric, frame x height x width x depth
        --> unit : text

        --------------------------------------
        links
        --------------------------------------
        imaging_space : VolumetricImageSpace
    }

    class ImagingSpace{
        <<NWBContainer>>

        --------------------------------------
        datasets
        --------------------------------------
        description : text
        origin_coordinates : numeric, length 3, optional
        --> unit : text, default="micrometers"

        --------------------------------------
        attributes
        --------------------------------------
        location : text, optional
    }

    class PlanarImagingSpace{
        <<ImagingSpace>>

        --------------------------------------
        datasets
        --------------------------------------
        grid_spacing : numeric, length 2, optional
        --> unit : text, default="micrometers"

        --------------------------------------
        attributes
        --------------------------------------
        reference_frame : text, optional
    }

    class VolumetricImagingSpace{
        <<ImagingSpace>>

        --------------------------------------
        datasets
        --------------------------------------
        grid_spacing : numeric, length 2, optional
        --> unit : text, default="micrometers"

        --------------------------------------
        attributes
        --------------------------------------
        reference_frame : text, optional
    }

    class MicroscopyTable{
        <<DynamicTable>>

        --------------------------------------
        datasets
        --------------------------------------
        VectorData :
        - location : text, optional
        - coordinates : numeric, length 2, optional
        --> unit : text, default="micrometers"
        - indicator : Indicator
        - notes : text, optional
        - microscope : Microscope
        - excitation_source : ExcitationSource
        - photodetector : Photodetector
        - dichroic_mirror : DichroicMirror
        - emission_filter : OpticalFilter
        - excitation_filter : OpticalFilter
        - objective_lens : ObjectiveLens
    }

    class Microscopy{
        <<LabMetaData>>

        --------------------------------------
        groups
        --------------------------------------
        microscopy_table : MicroscopyTable
    }

    PlanarMicroscopySeries *-- MicroscopySeries : extends
    PlanarMicroscopySeries -- PlanarImagingSpace : links
    VariableDepthMicroscopySeries *-- MicroscopySeries : extends
    VariableDepthMicroscopySeries -- PlanarImagingSpace : links
    VolumetricMicroscopySeries *-- MicroscopySeries : extends
    VolumetricMicroscopySeries -- VolumetricImagingSpace : links
    PlanarImagingSpace *-- ImagingSpace : extends
    VolumetricImagingSpace *-- ImagingSpace : extends
    MicroscopySeries ..> MicroscopyTable : reference rows
    Microscopy ..> MicroscopyTable : links
```

- [x] add diagram schema 
- [x] add mock functions for Microscopy and MicroscopyTable

### CODE REVIEW

File: src/pynwb/ndx_microscopy/testing/_mock.py [hunk 1]
```diff
@@ -7,73 +7,32 @@
...
+    description: str = "This is a mock instance of a Microscopy type to be used for rapid testing.",
+) -> ndx_microscopy.Microscopy:
+    microscopy = ndx_microscopy.Microscopy(
+        name=name or name_generator("Microscopy"),
         description=description,
-        manufacturer=manufacturer,
-        model=model,
     )
-    return microscope
+    return microscopy
```
Comment (CodyCBakerPhD) [inline src/pynwb/ndx_microscopy/testing/_mock.py:19]: Did you mean to do this?

File: spec/ndx-microscopy.namespace.yaml [hunk 2]
```diff
@@ -8,6 +8,8 @@ namespaces:
   - cody.baker@catalystneuro.com
   - alessandra.trapani@catalystneuro.com
   schema:
-  - namespace: core
+  - namespace:
+    - core
+    - ndx-ophys-devices
```
Comment (CodyCBakerPhD) [inline spec/ndx-microscopy.namespace.yaml:13]: I was told we would need something like this

@rly does this look right? 

w.r.t. https://github.com/catalystneuro/ndx-ophys-devices

Comment (rly) [inline spec/ndx-microscopy.namespace.yaml:13]: Should be:
```suggestion
  - namespace: core
  - namespace: ndx-ophys-devices
```

`schema` should be a list of dicts with certain keys

Comment (rly) [reply spec/ndx-microscopy.namespace.yaml:13]: In general, I recommend using the API to create the specs, though I know it adds another step.

File: src/pynwb/ndx_microscopy/testing/_mock.py (hunk 1 above)
Comment (alessandratrapani) [reply src/pynwb/ndx_microscopy/testing/_mock.py]: No sorry, I forgot to put a TODO comment, it was a work in progress

**Comment by CodyCBakerPhD** [comment]
@alessandratrapani Two small conflicts

**Comment by CodyCBakerPhD** [comment]
replaced by #22

**Pull request closed without merging**